lib/redcord/attribute.rb

```diff
@@ -54,6 +65,24 @@ def index_attribute(attr, type)
         class_variable_get(:@@index_attributes) << attr
       end
     end
+    
+    sig { params(index_name: Symbol, attrs: T::Array[Symbol]).void }
+    def custom_index(index_name, attrs)
+      attrs.each do |attr|
+        type = props[attr][:type]
+        if !can_custom_index?(type)
+          raise(Redcord::WrongAttributeType, "Custom index doesn't support '#{type}' attributes.")
+        end
+      end
+      shard_by_attr = class_variable_get(:@@shard_by_attribute)
+      if shard_by_attr and shard_by_attr != attrs.first
```

Comment:
This is not a necessary limitation, I agree. The motivation is to prevent user from implementing sub optimal indexing.
Consider you have custom index: [:a, :b]

If you shard by :b, you only can query: 
- :a (value) & :b (value). 
You have to include :b and can't query :b by range.
You limit your options and also don't get much of performance improvement compared to using single-attribute indexes.

If you shard by :a, you can do more:
- :a (value)
- :a (value) & :b (range or value)

If you believe we should let user implement it sub optimally, I'm fine with removing this limitation.